src/Resources/config/core.php

```diff
@@ -78,11 +78,13 @@
             ->tag('routing.loader')
             ->args([
                 new ReferenceConfigurator('sonata.admin.pool'),
-                [],
-                new ReferenceConfigurator('service_container'),
             ])
 
         ->alias(AdminPoolLoader::class, 'sonata.admin.route_loader')
+            ->deprecate(...BCDeprecationParameters::forConfig(
```

Comment:
I've also deprecated the alias since I don't think it's useful, the original service is meant to be loaded as `routing.loader`, not to inject it in another service.